Source/loadsave.cpp

```diff
@@ -1623,9 +1619,9 @@ void LoadHeroItems(Player &player)
 
 	gbIsHellfireSaveGame = file.NextBool8();
 
-	LoadMatchingItems(&file, NUM_INVLOC, player.InvBody);
-	LoadMatchingItems(&file, NUM_INV_GRID_ELEM, player.InvList);
-	LoadMatchingItems(&file, MAXBELTITEMS, player.SpdList);
+	LoadMatchingItems(file, NUM_INVLOC, player.InvBody);
+	LoadMatchingItems(file, NUM_INV_GRID_ELEM, player.InvList);
+	LoadMatchingItems(file, MAXBELTITEMS, player.SpdList);

```

Comment:
You removed the iteration constants related to `LoadItems` above, but we are doing something very similar here.

Can't we leverage some of the iterator work created recently (to iterate on all "valid" player items) for this as well, by creating a variation of that that iterates on _all_ items?

That would combine the iteration here, and on the other loops, as well as remove all of these constants from the equation.

Reply:
From the description this looks to be prep for doing just that.

Reply:
Yeah, once I hit 500 changed lines I figured it was worth getting feedback on. These loops are more involved than the simple cases, and because `InvBody` et al are C-style arrays instead of STL containers using them in a function requires passing the size information as a separate parameter. I could change `LoadMatchingItems` to operate on a Player reference but I figured it would be better to leave it till we can use containers for the collections instead.